lib/eclipse/EclipseState/Schedule/MSW/SpiralICD.cpp

```diff
@@ -0,0 +1,72 @@
+/*
+  Copyright 2017 SINTEF Digital, Mathematics and Cybernetics.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <opm/parser/eclipse/EclipseState/Schedule/MSW/SpiralICD.hpp>
+#include <opm/parser/eclipse/Deck/DeckRecord.hpp>
+#include <opm/parser/eclipse/Deck/DeckKeyword.hpp>
+
+#include <limits>
+
+
+namespace Opm {
+
+    SpiralICD::SpiralICD(const DeckRecord& record)
+    {
+        strength = record.getItem("STRENGTH").getSIDouble(0);
+        length = record.getItem("LENGTH").getSIDouble(0);
+        density_calibration = record.getItem("DENSITY_CALI").getSIDouble(0);
+        viscosity_calibration = record.getItem("VISCOSITY_CALI").getSIDouble(0);;
+        critical_value = record.getItem("CRITICAL_VALUE").getSIDouble(0);
+        width_transition_region = record.getItem("WIDTH_TRANS").get< double >(0);
+        max_viscosity_ratio = record.getItem("MAX_VISC_RATIO").get< double >(0);
+        method_flow_scaling = record.getItem("METHOD_SCALING_FACTOR").get < int >(0);
+        if (record.getItem("MAX_ABS_RATE").hasValue(0) )
```

Comment:
OK; I guess you could add a default value of `1e-20` for the `MAX_ABS_RATE` item in the `WEGSICD` keyword definition - then everything here is *pure assignment* and the full constructor could be implemented with an initializer list(?):

```C++
SpiralICD::SpiralICD(const DeckRecord& record) :
   stringth( record.getItem("STRENGTH").getSIDouble(0)),
   length( record.getItem("LENGTH").getSiDouble(0)), 
    ...
    status( record.getItem("STATUS").getTrimmedString(0))
{  }
```

Reply:
`1.e-20` will not work, since by default it should be no limit. `1.e-20` will give a too strong limit here. 

I can do the following, 
```
max_absolute_rate(record.getItem("MAX_ABS_RATE").hasValue(0) ? record.getItem("MAX_ABS_RATE").getSIDouble(0)  : std::numeric_limits<double>::lowest(); ```
```

Maybe `lowest()` should have been `max()`. I was thinking a negative value can be used to avoid comparison. It really depends on how to enforce `No limit` here.

Reply:
Ahhh I misunderstood; I interpreted `lowest` as *closest to zero*; but then I would say you should use `max( )` instead of `lowest( )`.